### PR TITLE
feat: test Safe on Sepolia & Pimlico

### PR DESCRIPTION
**Pull request by chris13524**
Tests the Safe code against Sepolia and Pimlico

E.g. transaction: https://sepolia.etherscan.io/address/0x07422e736d74dbb0b666cb9dbd6424d9173f183d#internaltx

Screenshot 2024-09-11 at 13 41 52

Screenshot 2024-09-11 at 13 42 03

### CODE REVIEW

File: crates/yttrium/src/transaction/send/safe_test.rs [hunk 1]
```diff
@@ -449,11 +450,40 @@ mod tests {
...
-        let rpc_url: reqwest::Url = rpc_url.parse()?;
-        let provider = ReqwestProvider::<Ethereum>::new_http(rpc_url);
+    async fn use_faucet(
+        provider: ReqwestProvider,
+        faucet: LocalSigner<SigningKey>,
+        amount: U256,
+        to: Address,
+    ) -> eyre::Result<()> {
+        if amount > U256::from(20) {
+            panic!("You probably don't need that much");
```
Comment (chris13524) [inline crates/yttrium/src/transaction/send/safe_test.rs:460]: Basic check which we can tune to make sure we don't use excessive amount of test ETH. It is not free, so we should use the minimum amount necessary.

File: README.md [hunk 2]
```diff
@@ -87,4 +87,35 @@ After installing the dependencies, clone the repository and run the following co
...
 
-This will fetch the third party dependencies and build the project, including the Swift bindings.
\ No newline at end of file
+This will fetch the third party dependencies and build the project, including the Swift bindings.
+
+### Devloop
+
+During normal development you can use the `just devloop` command to test your code both during development and before comitting/pushing. This is handy as it runs as many checks as possible and fixes any issues (such as formatting) automatically.
+
+This command does not require any configuration.
```
Comment (jackpooleywc) [inline README.md:96]: We need to add `just` and `pnpm` to the list of dependencies above ☝️ we should possibly just (no pun intended) remove the `Makefile` and move all the commands to the `justfile`

File: crates/yttrium/src/transaction/send/safe_test.rs [hunk 3]
```diff
@@ -449,11 +450,40 @@ mod tests {
...
+                TransactionRequest::default().with_to(to).with_value(amount),
+            )
+            .await?
+            .watch()
+            .await?;
+        let balance = provider.get_balance(to).await?;
+        assert_eq!(balance, amount);
+
+        Ok(())
+    }
```
Comment (jackpooleywc) [inline crates/yttrium/src/transaction/send/safe_test.rs:478]: nice! we should share this with all tests I think 💯

**Review by jackpooleywc — APPROVED** [review]
:shipit:

File: crates/yttrium/src/transaction/send/safe_test.rs (hunk 3 above)
Comment (chris13524) [reply crates/yttrium/src/transaction/send/safe_test.rs:478]: Yeah welcome to use it!

File: README.md (hunk 2 above)
Comment (chris13524) [reply README.md:101]: Added in 5831cbaeae4ad76ef4a5c9988afbab562ab2713f

> we should possibly just (no pun intended) remove the Makefile and move all the commands to the justfile

Yeah that would be nice

**Pull request merged**